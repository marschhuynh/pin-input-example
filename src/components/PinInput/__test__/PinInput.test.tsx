import { render, waitFor } from '@testing-library/react';
import { PinInput } from '../index';

describe('PinInput', () => {
  it('matches snapshot with default value', async () => {
    const { baseElement } =
      render(<PinInput
        defaultValue='12'
      />)
    // eslint-disable-next-line testing-library/no-wait-for-snapshot
    await waitFor(() => expect(baseElement).toMatchSnapshot())
  })
  it('matches snapshot with secret mode', async () => {
    const { baseElement } =
      render(<PinInput
        defaultValue='12'
        isSecretMode
      />)
    // eslint-disable-next-line testing-library/no-wait-for-snapshot
    await waitFor(() => expect(baseElement).toMatchSnapshot())
  })
})

