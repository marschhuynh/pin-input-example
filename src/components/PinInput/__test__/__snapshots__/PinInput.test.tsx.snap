// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PinInput matches snapshot with default value 1`] = `
<body>
  <div>
    <div
      class="sc-hLBbgP cpoLxT"
    >
      <input
        class="sc-gswNZR hMcZeN"
        id=":r0:"
        value="12"
      />
      <label
        class="sc-dkrFOg hwNuws"
        for=":r0:"
      >
        <div
          class="sc-bcXHqe ckpZhT number-item"
        >
          1
        </div>
        <div
          class="sc-bcXHqe ckpZhT number-item"
        >
          2
        </div>
        <div
          class="sc-bcXHqe ckpZhT number-item active"
        />
        <div
          class="sc-bcXHqe ckpZhT number-item"
        />
      </label>
    </div>
  </div>
</body>
`;

exports[`PinInput matches snapshot with secret mode 1`] = `
<body>
  <div>
    <div
      class="sc-hLBbgP cpoLxT"
    >
      <input
        class="sc-gswNZR hMcZeN"
        id=":r1:"
        value="12"
      />
      <label
        class="sc-dkrFOg hwNuws"
        for=":r1:"
      >
        <div
          class="sc-bcXHqe ckpZhT number-item"
        >
          *
        </div>
        <div
          class="sc-bcXHqe ckpZhT number-item"
        >
          *
        </div>
        <div
          class="sc-bcXHqe ckpZhT number-item active"
        />
        <div
          class="sc-bcXHqe ckpZhT number-item"
        />
      </label>
    </div>
  </div>
</body>
`;
